maliput/src/api/mod.rs

```diff
@@ -973,6 +973,15 @@ impl LaneSRange {
     }
 }
 
+/// A specific endpoint of a specific Lane.
+/// This is analogous to the C++ maliput::api::LaneEnd implementation.
+pub enum LaneEnd<'a> {
```

Comment:
I like this very much.

Reply:
Is it possible to constraint the creation such that no nullptr is used?

Reply:
No nullptr will be used.

Lane is a class whose underlying value is a reference.
The methods returning lanes from the underlying cpp implementation will evaluate whether that pointer is valid or not before creating the reference. 

We can meet and discuss if needed.

Btw, In a follow up PR (where I add LaneEndSet) I am changing the &Lane by Lane. You will notice why.

Reply:
https://github.com/maliput/maliput-rs/pull/62